plugins/autocomplete/plugin.js

```diff
@@ -0,0 +1,1422 @@
+		 */
+		getHtmlToInsert: function( item ) {
+			var encodedItem = encodeItem( item );
+			return this.outputTemplate ? this.outputTemplate.output( encodedItem ) : encodedItem.name;
+		},
+
+		/**
+		 * Creates and returns the model instance. This method is used while
+		 * initializing the autocomplete and can be overwritten in order to
+		 * return an instance of a different class than the default model.
+		 *
+		 * @param {Function} dataCallback See {@link CKEDITOR.plugins.autocomplete} arguments.
+		 * @returns {CKEDITOR.plugins.autocomplete.model} The model instance.
+		 */
+		getModel: function( dataCallback ) {
+			return new Model( dataCallback );
+		},
+
+		/**
+		 * Creates and returns the text watcher instance. This method is used while
+		 * initializing the autocomplete and can be overwritten in order to
+		 * return an instance of a different class than the default text watcher.
+		 *
+		 * @param {Function} textTestCallback See {@link CKEDITOR.plugins.autocomplete} arguments.
+		 * @returns {CKEDITOR.plugins.textWatcher} The text watcher instance.
+		 */
+		getTextWatcher: function( textTestCallback ) {
+			return new CKEDITOR.plugins.textWatcher( this.editor, textTestCallback, this.throttle );
+		},
+
+		/**
+		 * Creates and returns the view instance. This method is used while
+		 * initializing the autocomplete and can be overwritten in order to
+		 * return an instance of a different class than the default view.
+		 *
+		 * @returns {CKEDITOR.plugins.autocomplete.view} The view instance.
+		 */
+		getView: function() {
+			return new View( this.editor );
+		},
+
+		/**
+		 * Opens the panel if {@link CKEDITOR.plugins.autocomplete.model#hasData there is any data available}.
+		 */
+		open: function() {
+			if ( this.model.hasData() ) {
+				this.model.setActive( true );
+				this.view.open();
+				this.model.selectFirst();
+				this.view.updatePosition( this.model.range );
+			}
+		},
+
+		// LISTENERS ------------------
+
+		/**
+		 * The function registered by the {@link #attach} method as the
+		 * {@link CKEDITOR.editor#change} event listener.
+		 */
+		onChange: function() {
+			if ( this.model.isActive ) {
+				this.view.updatePosition( this.model.range );
+			}
+		},
+
+		/**
+		 * The function registered by the {@link #attach} method as the
+		 * {@link CKEDITOR.plugins.autocomplete.model#change-data} event listener.
+		 *
+		 * @param {CKEDITOR.eventInfo} evt
+		 */
+		onData: function( evt ) {
+			if ( this.model.hasData() ) {
+				this.view.updateItems( evt.data );
+				this.open();
+			} else {
+				this.close();
+			}
+		},
+
+		/**
+		 * The function registered by the {@link #attach} method as the
+		 * {@link CKEDITOR.plugins.autocomplete.view#click-item} event listener.
+		 *
+		 * @param {CKEDITOR.eventInfo} evt
+		 */
+		onItemClick: function( evt ) {
+			this.commit( evt.data );
+		},
+
+		/**
+		 * The function registered by the {@link #attach} method as the `keydown` event listener.
```

Comment:
Name of the method seems to be fine, but the description could be simplified. It should not mention where it is registered (there's a high chance someone will change it in the future and will miss this method).